processor/memorylimiter/memorylimiter.go

```diff
@@ -69,24 +78,27 @@ type memoryLimiter struct {
 
 // newMemoryLimiter returns a new memorylimiter processor.
 func newMemoryLimiter(logger *zap.Logger, cfg *Config) (*memoryLimiter, error) {
-	const mibBytes = 1024 * 1024
-	memAllocLimit := uint64(cfg.MemoryLimitMiB) * mibBytes
-	memSpikeLimit := uint64(cfg.MemorySpikeLimitMiB) * mibBytes
 	ballastSize := uint64(cfg.BallastSizeMiB) * mibBytes
 
 	if cfg.CheckInterval <= 0 {
 		return nil, errCheckIntervalOutOfRange
 	}
-	if memAllocLimit == 0 {
-		return nil, errMemAllocLimitOutOfRange
+	if cfg.MemoryLimitMiB == 0 && cfg.MemoryLimitPercentage == 0 {
+		return nil, errLimitOutOfRange
 	}
-	if memSpikeLimit >= memAllocLimit {
-		return nil, errMemSpikeLimitOutOfRange
+
+	decision, err := getDecision(cfg, logger)
+	if err != nil {
+		return nil, err
 	}
 
+	logger.Info("Percentage limiter configured",
```

Comment:
This is not accurate it can be the "fixed decision" one, since if the percentage it was already logged below here it can just say "Memory limiter configured".